STYLE.md

```diff
@@ -0,0 +1,432 @@
+
+General guide lines
+-------------------
+### Line Length
+
+Maximum line length is *80 characters* or *100 characters* if necessary.
+
+Modern screens have high definition and big width.
+But with some tiling managers and two terminals on one screen you are not able to
+see many characters on one line.
+On the other hand, restricting line size to a very small number like 80 leads to
+some crazy indentation despite the fact that
+shorter lines should force you to write well structured code.
+That's why *100* is a reasonable compromize.
+
+
+### Indentation
+
+Tabs are illegal.  Use spaces for indenting.  Indent your code blocks
+with *2 spaces*.  Indent the `where` keyword with two spaces to set it
+apart from the rest of the code and indent the definitions in a
+`where` clause with 2 spaces.  Some examples:
+
+```haskell
+sayHello :: IO ()
+sayHello = do
+    name <- getLine
+    putStrLn $ greeting name
+  where
+    greeting name = "Hello, " ++ name ++ "!"
+
+filter :: (a -> Bool) -> [a] -> [a]
+filter _ [] = []
+filter p (x:xs)
+  | p x       = x : filter p xs
+  | otherwise = filter p xs
+```
+
+### Blank Lines
+
+* One blank line between top-level definitions.
+* No blank lines between type signatures and function definitions.
+* Add one blank line between
+functions in a type class instance declaration if the function bodies
+are large.
+* You can add blank lines inside a big `do` block to separate logical
+parts of it.
+* You can also use blank lines to separate definitions inside `where`
+clause, if necessary.
+
+### Whitespace
+
+* Surround binary operators with a single space on either side. In case of
+currying add one space between the argument and the operation.
+
+* Use some tool to remove trailing spaces.
+
+### Naming convention
+
+Casing:
++ **_lowerCamelCase_** for function and variable names.
++ **_UpperCamelCase_** for types.
+
+* Don't use short names like `n`, `sk`, `f` unless their meaning is clear from
+context (function name, types, other variables, etc.).
+
+* For readability reasons, don't capitalize all letters when using an
+abbreviation.  For example, write `HttpServer` instead of
+`HTTPServer`.  Exception: two or three letter abbreviations, e.g. `IO`, `STM`.
+
+**Records name conventions**
+
+If data type has only one constructor then this data type name should be same
+as constructor name (also applies to `newtype`).
+
+```haskell
+data User = User Int String
+```
+TODO: Convention for the `newtype` field name?
+
+TODO: Convention for field names for record data types?
+
+e.g
+```haskell
+data NetworkConfig = NetworkConfig
+  { ncDelay :: Microsecond  -- `nc` corresponds to `_N_etwork_C_onfig`
+  , ncPort  :: Word
+  }
+```
+
+### Comments
+
+#### Punctuation
+
+Write proper sentences; start with a capital letter and use proper
+punctuation.
+
+#### Top-Level Definitions
+
+* Comment every top level function (particularly exported functions),
```

Comment:
I think that this is a very good recommendation for library code.

But on the other hand, in code that's at the top of the call stack, that's less of a concern, and sometimes leads to non-functional comments like in https://github.com/input-output-hk/cardano-node/blob/master/cardano-node/app/wallet-client.hs#L34 (no offense to the authors!).

Whatever the function does is fairly obvious from its name and type -- so I believe the comment is a waste of screen estate.

I think we should relax the wording.

Reply:
Updated.